File: Practica3/src/main/java/com/mycompany/Cola.py
class Cola:
    def __init__(self, n):
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = 0
        self.n = n
        self.count = 0

    def insert(self, e):
        if not self.isFull():
            self.arreglo[self.fin] = e
            self.fin = (self.fin + 1) % self.n
            self.count += 1
        else:
            raise OverflowError("Cola llena")

    def remove(self):
        if not self.isEmpty():
            valor = self.arreglo[self.inicio]
            self.inicio = (self.inicio + 1) % self.n
            self.count -= 1
            return valor
        else:
            raise IndexError("Cola vacía")

    def peek(self):
        if not self.isEmpty():
            return self.arreglo[self.inicio]
        else:
            raise IndexError("Cola vacía")

    def isEmpty(self):
        return self.count == 0

    def isFull(self):
        return self.count == self.n

    def size(self):
        return self.count

File: Practica3/src/main/java/com/mycompany/practica3/Cola.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.practica3;

/**
 *
 * @author Usuario
 */
public class Cola {

    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;
    private int count;

    public Cola(int n) {
        this.n = n;
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = 0;
        this.count = 0;
    }

    public void insert(long e) {
        if (!isFull()) {
            arreglo[fin] = e;
            fin = (fin + 1) % n;
            count++;
        } else {
            throw new RuntimeException("Cola llena");
        }
    }

    public long remove() {
        if (!isEmpty()) {
            long valor = arreglo[inicio];
            inicio = (inicio + 1) % n;
            count--;
            return valor;
        } else {
            throw new RuntimeException("Cola vacía");
        }
    }

    public long peek() {
        if (!isEmpty()) {
            return arreglo[inicio];
        } else {
            throw new RuntimeException("Cola vacía");
        }
    }

    public boolean isEmpty() {
        return count == 0;
    }

    public boolean isFull() {
        return count == n;
    }

    public int size() {
        return count;
    }
}
